odoo/tools/profiler.py

```diff
@@ -234,6 +234,7 @@ class SyncCollector(Collector):
     name = 'traces_sync'
 
     def start(self):
+        assert sys.gettrace() is None, "Using settrace would break coverage.py."
```

Comment:
This one is a little brutal mainly in stable, I would log an error and return just in case. 

Also, it would be possible to have a similar problem with nested SyncCollector, or if any other tool/code uses sys.settrace. So the message could be `"Cannot start SyncCollector, settrace already set: %s", sys.gettrace()`


Except that lgtm :) 



This one is a little brutal mainly in stable, I would log an error and return just in case. 

Also, it would be possible to have a similar problem with nested SyncCollector, or if any other tool/code uses sys.settrace. So the message could be `"Cannot start SyncCollector, settrace already set: %s", sys.gettrace()`


Except that lgtm :)

Reply:
Thanks for the review! I've also updated the message in `skipTest`. If it's not ok I can change it :)